File: 219-contains-duplicate-ii/contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& a, int k) 
    {
        unordered_set<int> set;
        int i=0;
        for(int j=0;j<a.size();j++)
        {
            if(set.count(a[j])>0)
            return true;
            set.insert(a[j]);
            if(j-i==k)
            {
                set.erase(a[i]);
                i++;
            }
        }
        return false;
    }
};

File: 219-contains-duplicate-ii/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) 
    {
        HashSet<Integer> set=new HashSet<>();
        int i=0;
        for(int j=0;j<nums.length;j++)
        {
            if(!set.add(nums[j]))
            return true;
            if(j-i==k)
            {
                set.remove(nums[i]);
                i++;
            }
        }
        return false;
    }
}